### ISSUE TITLE
Search bar doesn't search tags

### ISSUE THREAD

**Issue by mdosch** [jabbercat/jabbercat#17]
When using the search bar only the matching JIDs and nicknames are shown. You should also consider searching the tags. If you have a lot of contacts in the roster and none of the contacts in the visible area have the tag you want to filter for, you have to scroll first to be able to click the tag.

**Comment by horazont** [comment]
Note that there should be a filter button to the left of the search bar where you can select tags to filter for. (You might not see that if your icon theme is buggy / not compatible, which it is almost everywhere … Known issue, kindof. The button should still work even if it is invisible.)

But indeed, the search should include searching for tags and there should also be a way to convert typed tags into a proper filter (like in the drop-down menu on the filter button).

**Comment by horazont** [comment]
Yeah, I think we should replace the tag filter thing with directly searching for tag substrings. The tag filter thing is too cumbersome, even when using the mouse to click on tags. Also the code to allow clicking on tags in the roster is rather complex.

**Comment by sebastianriese** [comment]
Hmm, how about marking tag searches with ``#`` (and have tab completion/combo box behaviour in the search bar)? I think separating the search spaces for tags and jids/nicks is a Good Thing.

**Comment by horazont** [comment]
I have several "roster" entries (bookmarks) which start with ``#`` (IRC channels) though.

**Comment by sebastianriese** [comment]
Okay, that makes that idea a bad one, unless you want to introduce escaping with ``\`` (to get recursion of bad ideas).
